### PR TITLE
test(app launch profiling): wipe test data from disk for integration test

### PR DESCRIPTION
**Pull request by armcknight**
Following on from #3558 and followed by https://github.com/getsentry/sentry-cocoa/pull/3605

I had originally had the wipe-data option but thought I could remove it. Turns out it does help, if you run this test multiple times locally with failures, it helps to have a clean state each time. Also made it so the sample app removes the profile data persisted to a file, which only happens `#if defined(TEST)`

#skip-changelog

### CODE REVIEW

**Review by brustolin — APPROVED** [review]
LGTM

File: Samples/iOS-Swift/iOS-Swift/AppStartup.m [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#import "AppStartup.h"
+
+@implementation AppStartup
+
+// we must do this in objective c, because it's not permitted to be overridden in Swift
```
Comment (philipphofmann) [inline Samples/iOS-Swift/iOS-Swift/AppStartup.m:5]: Thanks for the comment 🥇

**Review by philipphofmann — APPROVED** [review]
LGTM

**Review by philipphofmann — APPROVED** [review]
Again LGTM.

**Pull request merged**